go/vt/topo/server.go

```diff
@@ -216,6 +216,21 @@ type Server interface {
 	// Can return ErrNoNode.
 	DeleteEndPoints(cell, keyspace, shard string, tabletType TabletType) error
 
+	// WatchEndPoints returns a channel that receives notifications
+	// every time EndPoints for the given type changes.
+	// It should receive a notification with the initial value fairly
+	// quickly after this is set. A value of nil means the Endpoints
+	// object doesn't exist or is empty. To stop watching this
+	// EndPoints object, close the stopWatching channel.
+	// If the underlying topo.Server encounters an error watching the node,
+	// it should retry on a regular basis until it can succeed.
+	// The initial error returned by this method is meant to catch
+	// the obvious bad cases (invalid cell, invalid tabletType, ...)
+	// that are never going to work. Mutiple notifications with the
+	// same contents may be sent (for instance when the serving graph
+	// is rebuilt, but the content hasn't changed).
+	WatchEndPoints(cell, keyspace, shard string, tabletType TabletType) (notifications <-chan *EndPoints, stopWatching chan<- struct{}, err error)
```

Comment:
This can cause failures on import, unless you micromanage that part. Do you want to do the 2-phase approach like I did for vschema?